src/main/kotlin/enums/Operator.kt

```diff
@@ -1,16 +1,28 @@
 package enums
 
-enum class Operator(val operator: String) {
-    PLUS("+"),
-    MINUS("-"),
-    MULTIPLY("*"),
-    DIVIDE("/");
+enum class Operator(val operator: String, val calculation: (Int, Int) -> Int) {
+    PLUS("+", { left, right -> left + right }),
+    MINUS("-", { left, right -> left - right }),
+    MULTIPLY("*", { left, right -> left * right }),
+    DIVIDE("/", {
+            left, right -> require(right != 0) { "${ErrorCode.ERROR_DIVIDER_CANNOT_ZERO.code }: 나누는 수는 0이 될 수 없습니다." }
+            left / right
+        }
+    );
 
     companion object {
-        fun getOperators(): List<String> {
-            return enumValues<Operator>().map {
-                it.operator
-            }.toList()
+        private val mapping = enumValues<Operator>().associate { it.operator to it.calculation }
+
+        fun getMapping(): Map<String, (Int, Int) -> Int> {
+            return mapping
+        }
```

Comment:
OOP 설계의 핵심은 객체간 메세지 요청/응답을 통해 대화를 하도록 설계하는 것입니다. 
즉, 내부 값을 외부로 노출시키기보단 내부에 캐싱되어있는 값에서 적절한 값을 찾아서 반환만 해줘도 될 것 같네요. 
```suggestion
        fun getMapping(symbol: String): Operator = mapping[symbol] ?: throw IllegalArgumentException(...)
```

혹은 이펙티브 코틀린에서 제시하는 방식 중 예외를 던지기보단 예외를 적절히 다루라는 내용을 토대로
none enum으로 null반환 혹은 throw Exception을 하기보단, None enum타입을 만들어 제공할 수도 있습니다.